specification/confidentialledger/data-plane/ConfidentialLedger/models.tsp

```diff
@@ -0,0 +1,933 @@
+   * given claim data.
+   */
+  protocol: ApplicationClaimProtocol;
+}
+
+/**
+ * The contents of a receipt
+ */
+model ReceiptContents {
+  /**
+   * Certificate
+   */
+  cert?: string;
+
+  /**
+   * Leaf
+   */
+  leaf?: string;
+
+  /**
+   * Leaf components of the receipt
+   */
+  leafComponents?: ReceiptLeafComponents;
+
+  /**
+   * Node identifier
+   */
+  nodeId: string;
+
+  /**
+   * Proof
+   */
+  proof: ReceiptElement[];
+
+  /**
+   * Root
+   */
+  root?: string;
+
+  /**
+   * Service endorsements
+   */
+  serviceEndorsements?: string[];
+
+  /**
+   * The signature of the receipt
+   */
+  signature: string;
+}
+
+/**
+ * Components that make up a receipt leaf node.
+ */
+model ReceiptLeafComponents {
+  /**
+   * Claims digest
+   */
+  claimsDigest?: string;
+
+  /**
+   * Commit evidence
+   */
+  commitEvidence?: string;
+
+  /**
+   * TWrite set digest
+   */
+  writeSetDigest?: string;
+}
+
+/**
+ * Receipt element
+ */
+model ReceiptElement {
+  /**
+   * Left
+   */
+  left?: string;
+
+  /**
+   * Right
+   */
+  right?: string;
+}
+
+/**
+ * Response returned to a query for the transaction status
+ */
+model TransactionStatus {
+  /**
+   * Represents the state of the transaction.
+   */
+  state: TransactionState;
+
+  /**
+   * A unique identifier for the state of the ledger. If returned as part of a
+   * LedgerEntry, it indicates the state from which the entry was read.
+   */
+  transactionId: string;
+}
+
+/**
+ * Paginated users returned in response to a query.
+ */
+model PagedUsers {
+  /**
+   * Ledger users.
+   */
+  @pageItems
+  ledgerUsers?: LedgerUser[];
+
+  /**
+   * Path from which to retrieve the next page of results.
+   */
+  @nextLink
+  nextLink?: string;
+}
+
+/**
+ * Details about a Confidential Ledger user.
+ */
+model LedgerUser {
+  /**
+   * Represents an assignable role.
+   */
+  assignedRole: ConfidentialLedgerUserRoleName;
+
+  /**
+   * Identifier for the user. This must either be an AAD object id or a certificate
+   * fingerprint.
+   */
+  @visibility(Lifecycle.Read)
+  userId?: string;
+}
+
+/**
+ * Paginated users returned in response to a query.
+ */
+model PagedLedgerUsers {
+  /**
+   * Ledger users with details.
+   */
+  @pageItems
+  ledgerUsers?: LedgerUserMultipleRoles[];
+
+  /**
+   * Path from which to retrieve the next page of results.
+   */
+  @nextLink
+  nextLink?: string;
+}
+
+/**
+ * Details about a Confidential Ledger user.
+ */
+model LedgerUserMultipleRoles {
+  /**
+   * Represents an assignable role.
+   */
+  assignedRoles: ConfidentialLedgerUserRoleName[];
+
+  /**
+   * Identifier for the user. This must either be an AAD object id or a certificate
+   * fingerprint.
+   */
+  @visibility(Lifecycle.Read)
+  userId?: string;
+}
+
+/**
+ * bundle for the user defined endpoints
+ */
+model Bundle {
+  /**
+   * Metadata information for the bundle.
+   */
+  metadata: Metadata;
+
+  /**
+   * Any object
+   */
+  modules: ModuleDef[];
```

Comment:
The original swagger defined the type as:

```
"ModuleDef_array": {
      "items": {
        "$ref": "#/definitions/ModuleDef"
      },
      "type": "object"
    },
```
Please confirm if the typespec definition is accurate